packages/gatsby-plugin-sharp/src/index.js

```diff
@@ -555,8 +555,107 @@ async function resolutions({ file, args = {} }) {
   }
 }
 
+async function notMemoizedtraceSVG({ file, args, fileArgs }) {
+  const potrace = require(`potrace`)
+  const trace = Promise.promisify(potrace.trace)
+  let defaultArgs = {
+    color: `lightgray`,
+    optTolerance: 0.4,
+    turdSize: 100,
+    turnPolicy: `majority`,
+  }
+
+  const defaultFileResizeArgs = {
+    width: 400,
+    quality: 50,
+    jpegProgressive: true,
+    pngCompressionLevel: 9,
+    grayscale: false,
+    duotone: false,
+    toFormat: ``,
+  }
+  const options = _.defaults(fileArgs, defaultFileResizeArgs)
+  let pipeline = sharp(file.absolutePath).rotate()
+
+  pipeline
+    .resize(options.width, options.height)
+    .crop(options.cropFocus)
+    .png({
+      compressionLevel: options.pngCompressionLevel,
+      adaptiveFiltering: false,
+      force: args.toFormat === `png`,
+    })
+    .jpeg({
+      quality: options.quality,
+      progressive: options.jpegProgressive,
+      force: args.toFormat === `jpg`,
+    })
+
+  // grayscale
+  if (options.grayscale) {
+    pipeline = pipeline.grayscale()
+  }
+
+  // rotate
+  if (options.rotate && options.rotate !== 0) {
+    pipeline = pipeline.rotate(options.rotate)
+  }
+
+  // duotone
+  if (options.duotone) {
+    pipeline = await duotone(options.duotone, file.extension, pipeline)
+  }
+
+  const tmpDir = require(`os`).tmpdir()
+  const tmpFilePath = `${tmpDir}/${file.name}-${crypto
+    .createHash(`md5`)
+    .update(JSON.stringify(fileArgs))
+    .digest(`hex`)}.${file.extension}`
+
+  await new Promise(resolve =>
+    pipeline.toFile(tmpFilePath, (err, info) => {
+      resolve()
+    })
+  )
+
+  return trace(tmpFilePath, options)
+    .then(svg => optimize(svg))
+    .then(svg => encodeOptimizedSVGDataUri(svg))
+}
+
+const memoizedTraceSVG = _.memoize(
+  notMemoizedtraceSVG,
+  ({ file, args }) => `${file.absolutePath}${JSON.stringify(args)}`
+)
+
+async function traceSVG(args) {
+  return await memoizedTraceSVG(args)
+}
+
+// https://codepen.io/tigt/post/optimizing-svgs-in-data-uris
+function encodeOptimizedSVGDataUri(svgString) {
+  var uriPayload = encodeURIComponent(svgString) // encode URL-unsafe characters
+    .replace(/%0A/g, ``) // remove newlines
+    .replace(/%20/g, ` `) // put spaces back in
+    .replace(/%3D/g, `=`) // ditto equals signs
+    .replace(/%3A/g, `:`) // ditto colons
+    .replace(/%2F/g, `/`) // ditto slashes
+    .replace(/%22/g, `'`) // replace quotes with apostrophes (may break certain SVGs)
+
+  return `data:image/svg+xml,` + uriPayload
+}
+
+const optimize = svg => {
+  const SVGO = require(`svgo`)
+  const svgo = new SVGO()
```

Comment:
This resize technique above is super awesome!
It might also be worth looking into multipass, to compress the svg even further: 

```javascript
const svgo = new SVGO({ multipass: true, floatPrecision: 1 })
```

Reply:
Awww thanks Emil! 🤗 🙌

Reply:
Demo images on a diet after that (missed checking one, but here's the before/after for two of the three at https://deploy-preview-2456--using-gatsby-image.netlify.com/traced-svg/):

7.401 -> 4.506 Byte
13.629 -> 8.562 Byte

Thank you again Emil!

(Hey I could swear I didn't see it in https://github.com/EmilTholin/image-trace-loader when I checked … ah yes, only saw the initial version. We're lucky you are watching here!)

Reply:
Looking at image-trace-loader again, I hadn't noticed https://github.com/EmilTholin/image-trace-loader/blob/master/lib/index.js#L32-L57 either 😊 ❤️  … watchu think about that @KyleAMathews? (Been 👀 https://github.com/akfish/node-vibrant btw.)